database/src/main/java/com/firebase/ui/database/ObservableSnapshotArray.java

```diff
@@ -66,6 +71,11 @@ public ChangeEventListener addChangeEventListener(@NonNull ChangeEventListener l
     @CallSuper
     public void removeChangeEventListener(@NonNull ChangeEventListener listener) {
         mListeners.remove(listener);
+
+        // Reset mHasDataChanged if there are no more listeners
+        if (!isListening()) {
+            mHasDataChanged = false;
```

Comment:
This block is a little surprising to me.  Wouldn't you want the first listener attached to still fire immediately if the FirebaseArray contains data?